src/main.rs

```diff
@@ -504,6 +504,10 @@ fn verify_files_main(
     server: &mut dyn Server,
     user_files: &[FileId],
 ) -> Result<bool, VerifyError> {
+    // Limit errors or interrupts are saved here. Then rest of the procs are marked as unknown.
```

Comment:
While this seems to work, I find this somewhat error-prone and ugly. What do you think of this idea: We first register all source units in the server and mark them as unprocessed/unfinished/WIP. Then, when one proc results in a timeout, it is a) easy to mark all others as unknown in the UI, b) easy to identify the culprit (by distinguishing timeout vs unfinished), c) impossible to forget handling the LimitErrors/Interrupted in each loop. 

Hopefully that idea can avoid all of those extra `match` statements that were necessary to introduce in this patch.